src/provider/index.ts

```diff
@@ -139,14 +147,33 @@ export class Main {
     }
 
     private async addIntentListener(payload: IntentListenerPayload, identity: ProviderIdentity): Promise<void> {
-        const appWindow = this._model.getWindow(identity);
-        if (appWindow) {
-            appWindow.addIntentListener(payload.intent);
-            return;
-        } else {
-            // TODO: [SERVICE-429] Should this be an FDC3Error? Or is the AppWindow model an implementation detail and is OK to throw a non-FDC3 error?
-            throw new Error('App not found in model');
+        let appWindow = this._model.getWindow(identity);
+
+        // Window is not in model - this should mean that the app is not in the directory, as directory apps are immediately added to model upon window creation
```

Comment:
Mmhh not sure I follow. The app/window calling `addIntentListener` surely has imported the client?

Or do you mean the app/window that is the target of `raiseIntent`? In which case I think the "wait for intent handshake" in `fireIntent` would cover this?

Reply:
Sorry, my bad. Was assuming that `identity` could be an app that isn't using the FDC3 client - but since this method only runs if the service receives a message _from_ identity, then that window must have the client imported.